### ISSUE TITLE
Custom Rule not working

### ISSUE THREAD

**Issue by kalmeshnfk** [realm/SwiftLint#2507]
### Describe the bug
Added a Custom Rule for all hardcoded strings in the code except color codes that start with #. However, its not working.
 
```yml
# insert yaml contents here

whitelist_rules:
- custom_rules

excluded: # paths to ignore during linting. Takes precedence over `included`.
 - Pods

line_length: 300

custom_rules:
  no_hard_coded_strings:
    regex: '^[^#][a-zA-Z0-9][a-zA-Z0-9! ]+'
    match_kinds:
        - string
    message: "Hard-coded string is never a good practice"
    severity: warning

**Comment by jpsim** [comment]
I forget exactly the supposed behavior of `whitelist_rules` with `custom_rules`, but you might need to add your custom rule to the whitelist. Can you try that and report back?

```yaml
whitelist_rules:
  - custom_rules
  - no_hard_coded_strings

excluded: # paths to ignore during linting. Takes precedence over `included`.
 - Pods

line_length: 300

custom_rules:
  no_hard_coded_strings:
    regex: '^[^#][a-zA-Z0-9][a-zA-Z0-9! ]+'
    match_kinds:
        - string
    message: "Hard-coded string is never a good practice"
    severity: warning
```

**Comment by biboran** [comment]
Was this by any chance a nested configuration? Due to the way [configuration works](https://github.com/realm/SwiftLint/blob/d62e187b5c47978b1f33b89f55f3ba44367acc4b/Source/SwiftLintFramework/Models/Configuration.swift#L223-L232) I don't think it's possible to whitelist or disable one of the custom rules in the root configuration. If it is a nested configuration then #2532 might help. If it is a root configuration you would have to write an explicit `CustomRules` filtering in the `enableRules` function.

**Comment by parascse** [comment]
Hi @jpsim 

Thanks for identifying the issue with 'no_hard_coded_strings' rule. I am using 'opt_in_rules' in my lint file and I don't see this rule working with it, do you have any solution for that?
